cmake/partition_manager.cmake

```diff
@@ -27,231 +21,298 @@ macro(add_region)
-    NAME flash_primary
-    SIZE ${flash_size}
-    BASE ${CONFIG_FLASH_BASE_ADDRESS}
-    PLACEMENT complex
-    DEVICE NRF_FLASH_DRV_NAME
+    NAME otp
+    SIZE ${otp_size}
+    BASE ${otp_start_addr}
+    PLACEMENT start_to_end
     )
+endif()
 
-  if (CONFIG_PM_EXTERNAL_FLASH)
-    add_region(
-      NAME external_flash
-      SIZE ${CONFIG_PM_EXTERNAL_FLASH_SIZE}
-      BASE ${CONFIG_PM_EXTERNAL_FLASH_BASE}
-      PLACEMENT start_to_end
-      DEVICE ${CONFIG_PM_EXTERNAL_FLASH_DEV_NAME}
-      )
-  endif()
+math(EXPR flash_size "${CONFIG_FLASH_SIZE} * 1024" OUTPUT_FORMAT HEXADECIMAL)
 
-  set(pm_cmd
-    ${PYTHON_EXECUTABLE}
-    ${NRF_DIR}/scripts/partition_manager.py
-    --input-files ${input_files}
-    --regions ${regions}
-    --output-partitions ${CMAKE_BINARY_DIR}/partitions.yml
-    --output-regions ${CMAKE_BINARY_DIR}/regions.yml
-    ${static_configuration}
-    ${region_arguments}
-    )
+add_region(
+  NAME flash_primary
+  SIZE ${flash_size}
+  BASE ${CONFIG_FLASH_BASE_ADDRESS}
+  PLACEMENT complex
+  DEVICE NRF_FLASH_DRV_NAME
+  )
 
-  set(pm_output_cmd
-    ${PYTHON_EXECUTABLE}
-    ${NRF_DIR}/scripts/partition_manager_output.py
-    --input-partitions ${CMAKE_BINARY_DIR}/partitions.yml
-    --input-regions ${CMAKE_BINARY_DIR}/regions.yml
-    --config-file ${CMAKE_BINARY_DIR}/pm.config
-    --input-names ${images}
-    --header-files
-    ${header_files}
+if (CONFIG_PM_EXTERNAL_FLASH)
+  add_region(
+    NAME external_flash
+    SIZE ${CONFIG_PM_EXTERNAL_FLASH_SIZE}
+    BASE ${CONFIG_PM_EXTERNAL_FLASH_BASE}
+    PLACEMENT start_to_end
+    DEVICE ${CONFIG_PM_EXTERNAL_FLASH_DEV_NAME}
     )
+endif()
 
-  # Run the partition manager algorithm.
-  execute_process(
-    COMMAND
-    ${pm_cmd}
-    RESULT_VARIABLE ret
-    )
+set(pm_out_partition_files ${CMAKE_BINARY_DIR}/partitions_${domain}.yml)
+set(pm_out_region_files ${CMAKE_BINARY_DIR}/regions_${domain}.yml)
+set(pm_out_dotconf_files ${CMAKE_BINARY_DIR}/pm_${domain}.config)
+
+set(pm_cmd
+  ${PYTHON_EXECUTABLE}
+  ${NRF_DIR}/scripts/partition_manager.py
+  --input-files ${input_files}
+  --regions ${regions}
+  --output-partitions ${pm_out_partition_files}
+  --output-regions ${pm_out_region_files}
+  ${dynamic_partition_argument}
+  ${static_configuration}
+  ${region_arguments}
+  )
+
+set(pm_output_cmd
+  ${PYTHON_EXECUTABLE}
+  ${NRF_DIR}/scripts/partition_manager_output.py
+  --input-partitions ${pm_out_partition_files}
+  --input-regions ${pm_out_region_files}
+  --config-file ${pm_out_dotconf_files}
+  )
+
+# Run the partition manager algorithm.
+execute_process(
+  COMMAND
+  ${pm_cmd}
+  RESULT_VARIABLE ret
+  )
+
+if(NOT ${ret} EQUAL "0")
+  message(FATAL_ERROR "Partition Manager failed, aborting. Command: ${pm_cmd}")
+endif()
 
-  if(NOT ${ret} EQUAL "0")
-    message(FATAL_ERROR "Partition Manager failed, aborting. Command: ${pm_cmd}")
-  endif()
+# Produce header files and config file.
+execute_process(
+  COMMAND
+  ${pm_output_cmd}
+  RESULT_VARIABLE ret
```

Comment:
Should we read std out and error out, so that we can provide more feedback to user if this fails ?
https://cmake.org/cmake/help/latest/command/execute_process.html
`OUTPUT_VARIABLE`, `ERROR_VARIABLE`

Note: this is the danger of changing indentation in a PR. 
Feel free to create a followup task.

Reply:
The error message from python is being printed. 

I see the whole error trace, and I also see the failing command.

Reply:
great. Then no worries.